criu/util.c

```diff
@@ -1686,6 +1686,32 @@ ssize_t write_all(int fd, const void *buf, size_t size)
 	return n;
 }
 
+ssize_t pread_all(int fd, void *buf, size_t size, off_t offset)
+{
+	ssize_t n = 0;
+	while (size > 0) {
+		ssize_t ret = pread(fd, buf, size, offset);
+		if (ret == -1) {
+			if (errno == EINTR)
+				continue;
+			/*
+			 * The caller should use standard read() for
+			 * non-blocking I/O.
+			 */
+			if (errno == EAGAIN || errno == EWOULDBLOCK)
+				errno = EINVAL;
```

Comment:
This disturbs me, impact is not clear. Later we can process the code. Let's keep errno intact?

Reply:
This is actually a copy-paste of the `read_all` method above (just `read` -> `pread`). I don't see a problem in writing to errno?

Reply:
@AntonKozlov 
I agree with @rvansa - we could keep this consistent with other existing functions.
BTW I'd prefer using curly braces for one-line if's, but just a note.

Reply:
I would keep the @rvansa code. It is a copy-paste of `read_all` - while I would prefer its removal it would be a needless fork from upstream CRIU and it is not really required. If something then it could be rather an upstream CRIU contribution.
@wkia: CRIU does not use curly braces for one-line ifs: https://github.com/CRaC/criu/blob/crac/CONTRIBUTING.md#edit-the-source-code